src/migration.js

```diff
@@ -92,7 +105,8 @@ module.exports = class Migration {
    * @returns {boolean}
    */
   testFileName (needle) {
-    return this.file.indexOf(needle) === 0;
+    const formattedNeedle = this.options.migrations.nameFormatter(needle);
+    return this.file.indexOf(formattedNeedle) === 0;
```

Comment:
Are you sure calling `nameFormatter` here is necessary?

Reply:
That was the idea of @kf6kjg in his first response:
> However I'd go one step further and provide a way to format the name that is in storage too. That way I can take an existing database that was migrated without a formatter and still not have a migration collision. You could use the same format function, it would just force the user to add additional logic. Or you can have different functions for each side: one for formatting files from disk, another for formatting migration names from storage.

Reply:
Hmm... But is this `needle` (which I think is a terrible variable name by the way - not your fault of course) a full path? Or just a name? Because the `nameFormatter` function expects an absolute path as argument, right?

Reply:
`needle` is in this case the file name which was previously used for the migration but you can remove extensions etc. of course as well in the `nameFormatter`.

Reply:
I'm OK with this section of the code: a file name IS a relative path, thus any parser should behave itself.  My caveats are going to be in a different comment layer.